cargo-dist/src/env.rs

```diff
@@ -24,24 +24,37 @@ pub fn fetch_brew_env(
                 .arg("exec")
                 .arg("--")
                 .arg("/usr/bin/env")
+                // Splits outputs by NUL bytes instead of newlines.
+                // Ensures we don't get confused about env vars which
+                // themselves contain newlines.
+                .arg("-0")
                 .current_dir(working_dir)
                 .output()?;
 
-            return Ok(Some(String::from_utf8_lossy(&result.stdout).to_string()));
+            let s = String::from_utf8_lossy(&result.stdout).to_string();
+            // Split into lines based on the \0 separator
+            let output = s
+                // If there's a trailing NUL, we want that gone too
+                .trim_end_matches(&['\0', '\n'])
```

Comment:
whoops meant to be commenting on this

Reply:
I guess that's legal, and also extremely chaotic. Hm. I guess I could `trim_end_matches()` for the newline first, then `trim_end_matches` for `\0` as a second call. That'd avoid chomping a newline that comes before the `\0`.